lib/src/model/directus_item.dart

```diff
@@ -13,5 +14,8 @@ class DirectusItem {
   }
 
   /// Creates a new [DirectusItem]
-  DirectusItem(this._rawReceivedData);
+  DirectusItem(this._rawReceivedData)
+      : id = _rawReceivedData.containsKey("id")
```

Comment:
You are absolutely right. But to help user to use this package why not make DirectusItem and DirectusUser abstract also ?

I see some advantages to do so : 

- we can force the user to implement toMap() method and on other one like fromDirectus(Map<String, dynamic> rawReceivedData) which will return the desired type of object.
- it will be to be more consistent between DirectusItem and DirectusUser
- we also can move the endPoint string in DirectusItem as it is useless in DirectusUser, but we have to add a function which will return a Type object in a lot of methods.

What do you think about this ?

Reply:
I agree with you for `DirectusItem` that should be abstract, good idea.
`DirectusUser` cannot be abstract because we do create those instances ourselves when creating a new user, fetching users, etc.

The `toMap()` method could be made generic with the `updatedProperties` et `_rawReceivedData` maps we already have inside. We could just return the `_rawReceivedData` merged with the `updatedProperties` Map and this could be done in the `DirectusData` class so that it's available everywhere.

The `endPoint` does exist for users but since the methods are doing a very different things compared to item methods, you are correct we do not use it right now and it could be moved down.